src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -697,5 +718,55 @@ public void Dispose()
                 IOExtensions.DeleteFile(_tempFile);
             }
         }
+
+        public static bool ShouldResolveAttachmentsConflict(BlittableJsonReaderObject myMetadata, BlittableJsonReaderObject otherMetadata)
+        {
+            myMetadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray myAttachments);
+            otherMetadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray otherAttachments);
+            Debug.Assert(myAttachments != null || otherAttachments != null, "Cannot happen. We verified that we have a conflict in @attachments.");
+
+            var myAttachmentNames = new Dictionary<string, BlittableJsonReaderObject>();
```

Comment:
You mean case insensitive, right?
I made it case insensitive, but left it as dictionary as we do need the value later on to compare it.